lib/middleware/request_tracker.rb

```diff
@@ -273,7 +273,7 @@ def rate_limit(request)
           Discourse.warn("Global asset IP rate limit exceeded for #{ip}: 10 second rate limit", uri: request.env["REQUEST_URI"])
         end
 
-        return !(GlobalSetting.max_reqs_per_ip_mode == "warn")
+        return GlobalSetting.max_reqs_per_ip_mode != "warn" && limiter_assets10.seconds_to_wait(Time.now.to_i)
```

Comment:
That looks a bit weird. Hard to tell what `limiter_assets10.seconds_to_wait` returns and when is it `true` or `false`

Reply:
It should never be `true`. It's either `false` or a number representing the seconds until the request would no longer be blocked.